### PR TITLE
[ROCKETMQ-81] Add the RocketMq plugin for the Apache Spark

### PR DESCRIPTION
**Pull request by vesense**
https://issues.apache.org/jira/browse/ROCKETMQ-81

This is the init version for code review.
features:
* RocketMQReceiver - which is no fault-tolerance guarantees
* ReliableRocketMQReceiver - which is fault-tolerance guarantees

### CODE REVIEW

**Comment by vongosling** [comment]
@dongeforever @shroman @stevenschew what's your opinion about it

**Comment by vesense** [comment]
NOTICE: PR #5 has merged into this one, the work are completed by both @hustfxj and I. And I have added the credit for him too.

**Comment by zhouxinyu** [comment]
@vesense thanks, please @hustfxj help review this incorporative PR.

**Comment by hustfxj** [comment]
@vesense Thanks. Can you add add more tests for this plugin, especially the pull mode. Of course, it is difficult.

**Comment by vesense** [comment]
@hustfxj Yes, I will try to add more unit tests, especially pull mode.
@zhouxinyu Do you mind merging this firstly? I will add the tests maybe a few days later because of my time, and I don't want to block other peoples contributions.

File: rocketmq-storm/README.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# RocketMQ Storm Integration
```
Comment (shroman) [inline rocketmq-storm/README.md]: Is this pushed by mistake?

Comment (vesense) [inline rocketmq-storm/README.md]: No, this is just a placeholder for rocketmq-storm new `README.md` file. I plan to add some description in this file. It may be better to do this in another PR.

File: rocketmq-spark/src/main/scala/org/apache/rocketmq/spark/streaming/MQPullInputDStream.scala [hunk 2]
```diff
@@ -0,0 +1,535 @@
...
+
+  private val maxRateLimitPerPartition = optionParams.getOrDefault(RocketMQConfig.MAX_PULL_SPEED_PER_PARTITION,
+    "-1").toInt
+  
+  @transient private var kc: DefaultMQPullConsumer = null
+
+  /**
+    * start up timer thread to persis the OffsetStore
+    */
+  private val scheduledExecutorService = ThreadUtils.newDaemonSingleThreadScheduledExecutor(
```
Comment (hustfxj) [inline rocketmq-spark/src/main/scala/org/apache/rocketmq/spark/streaming/MQPullInputDStream.scala]: add @transient for this val

Comment (vesense) [inline rocketmq-spark/src/main/scala/org/apache/rocketmq/spark/streaming/MQPullInputDStream.scala]: @hustfxj Fixed.

**Comment by zhouxinyu** [comment]
Hi, I will merge this PR and close https://github.com/apache/incubator-rocketmq-externals/pull/5.

**Comment by zhouxinyu** [comment]
I have no idea about how to commit with two authors, so I thanked @hustfxj in the commit log, hope you don't mind ~~
